Control/lib/middleware/addDetectorId.middleware.js

```diff
@@ -0,0 +1,11 @@
+/**
+ * Middleware function to add detectorID to the request object
+ * @return {function(req, res, next): void} - middleware function
+ */
+const addDetectorIdMiddleware = (detectorId) => {
+  return async (req, res, next) => {
```

Comment:
You are missing some documentation from the return function. Please have a look `minimumRole.middleware.js` and try to improve the one here as well